File: <100/073/073.py
import sympy
import time

def main() -> None:
    max_d: int = 12000
    answer = 0
    # this algorithm, naive brute-force, is slow for Python; it takes 6 mins to execute
    # whereas C++ takes only 1 sec for the same algorithm.
    for d in range(max_d, 2, -1):
        # n/d < 1/2  <==> n < d/2  ==> max_n = d/2 - 1 if d%2 == 0 else d/2.
        n: int = (d//2 - 1) if d%2 == 0 else (d//2)
        while 3*n > d:
            answer += 1 if sympy.gcd(n, d) == 1 else 0
            n -= 1
    
    print(answer)
    return


if __name__ == '__main__':
    start = time.time()
    main()
    elapsed = time.time() - start

    print(f'{elapsed} sec')

File: <100/073/073.cpp
#include<iostream>
#include<numeric>
using namespace std;


int main(){
  const int max_d = 12000;
  int answer = 0;

  // naive brute-force;
  for(int d = max_d; d > 2; --d){
    // n/d < 1/2  <==> n < d/2  ==> max_n = d/2 - 1if d%2 == 0 else d/2.
    int n = d/2 - 1 + (d%2);
    while(3*n > d){
      answer += gcd(n, d) == 1;
      --n;
    }
  }

  cout << answer << endl;
  return 0;
}
